pkg/logs/input/docker/launcher.go

```diff
@@ -129,6 +141,8 @@ func (l *Launcher) run() {
 			containerID := removedService.Identifier
 			l.stopTailer(containerID)
 			delete(l.pendingContainers, containerID)
+		case containerId := <-l.erroredContainerID:
+			go l.restartTailer(containerId)
```

Comment:
What happens when we stop the agent and at least one restart routine is running ?

Reply:
It stops: the launcher will receive a stop signal and transmit it to all the started tailers
The new tailer to start won't be able to `setupReader` and dies quickly. Even if it could it would receive the `isClosedConnError` and would exit